sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ServiceRestClient.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace OpenTelemetry.Exporter.AzureMonitor
 {
-    internal partial class ServiceRestClient
+    internal partial class ServiceRestClient : IServiceRestClient
```

Comment:
This is an autogenerated rest client and will lose changes with autogen update. There is a way to create custom `ServiceRestClient` we may need to explore it.

Reply:
Hold that thought, i'm resolving conflicts. Looks like major changes to the RestClient.

I can sync with you offline to discuss more.
Here I'm changing a non-autogen class. 
If we can propose changes to the autogen class this won't be necessary. Please see my comment in IServiceRestClient.

Reply:
i abandoned this approach